homeassistant/components/jewish_calendar/__init__.py

```diff
@@ -5,104 +5,155 @@
+    name = config_entry.data.get(CONF_NAME, DEFAULT_NAME)
+    language = config_entry.data.get(CONF_LANGUAGE, DEFAULT_LANGUAGE)
+    diaspora = config_entry.data.get(CONF_DIASPORA, DEFAULT_DIASPORA)
+
+    if not config_entry.options:
+        # If options are not defined, update entry with optional values from the
+        # original configuration.
+        options = {
+            CONF_CANDLE_LIGHT_MINUTES: config_entry.data.get(
+                CONF_CANDLE_LIGHT_MINUTES, DEFAULT_CANDLE_LIGHT
+            ),
+            CONF_HAVDALAH_OFFSET_MINUTES: config_entry.data.get(
+                CONF_HAVDALAH_OFFSET_MINUTES, DEFAULT_HAVDALAH_OFFSET_MINUTES
+            ),
+        }
+
+        hass.config_entries.async_update_entry(config_entry, options=options)
+
+    # The current specification for config_entries of jewish_calendar allows to
+    # optionally specify latitude and longitude. So we can have a config_entry with
+    # CONF_LAT/LON specified, one with CONF_LOCATION or None in which case we'll
+    # take the Home location.
+    if CONF_LATITUDE in config_entry.data or CONF_LONGITUDE in config_entry.data:
+        latitude = config_entry.data.get(CONF_LATITUDE, hass.config.latitude)
+        longitude = config_entry.data.get(CONF_LONGITUDE, hass.config.longitude)
+    elif CONF_LOCATION in config_entry.data:
+        latitude = config_entry.data[CONF_LOCATION][CONF_LATITUDE]
+        longitude = config_entry.data[CONF_LOCATION][CONF_LONGITUDE]
+    else:
+        latitude = hass.config.latitude
+        longitude = hass.config.longitude
 
     location = Location(
+        name=hass.config.location_name,
+        diaspora=diaspora,
         latitude=latitude,
         longitude=longitude,
-        timezone=hass.config.time_zone,
-        diaspora=diaspora,
+        altitude=config_entry.data.get(CONF_ELEVATION, hass.config.elevation),
+        timezone=config_entry.data.get(CONF_TIME_ZONE, hass.config.time_zone),
     )
 
-    prefix = get_unique_prefix(
-        location, language, candle_lighting_offset, havdalah_offset
-    )
-    hass.data[DOMAIN] = {
-        "location": location,
+    hass.data.setdefault(DOMAIN, {})[config_entry.entry_id] = {
         "name": name,
         "language": language,
-        "candle_lighting_offset": candle_lighting_offset,
-        "havdalah_offset": havdalah_offset,
         "diaspora": diaspora,
-        "prefix": prefix,
+        "location": location,
+        "candle_lighting_offset": config_entry.options.get(
+            CONF_CANDLE_LIGHT_MINUTES, DEFAULT_CANDLE_LIGHT
+        ),
+        "havdalah_offset": config_entry.options.get(
+            CONF_HAVDALAH_OFFSET_MINUTES, DEFAULT_HAVDALAH_OFFSET_MINUTES
+        ),
     }
 
     for platform in PLATFORMS:
-        hass.async_create_task(async_load_platform(hass, platform, DOMAIN, {}, config))
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, platform)
+        )
```

Comment:
This can just be 1 call, please check some other core integrations for that functon